pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -107,6 +108,40 @@ private static String removeTerminatingSemicolon(String sql) {
     return sql;
   }
 
+  public static SqlNodeAndOptions compileToSqlNodeAndOptions(String sql)
+      throws Exception {
+    // Remove the comments from the query
+    sql = removeComments(sql);
+
+    // Remove the terminating semicolon from the query
+    sql = removeTerminatingSemicolon(sql);
+
+    // Extract OPTION statements from sql as Calcite Parser doesn't parse it.
+    List<String> options = extractOptionsFromSql(sql);
```

Comment:
this can be cleaned up by extending parser.jj to allow `OPTION` keyword but that can be done later.

Reply:
Yeah, since we already have the OPTIONS handler and be able to parse it out, then I don't complain. 
Eventually, this should all go to Parser.